purchases/src/main/kotlin/com/revenuecat/purchases/utils/EventsFileHelper.kt

```diff
@@ -0,0 +1,58 @@
+package com.revenuecat.purchases.utils
+
+import android.os.Build
+import androidx.annotation.RequiresApi
+import com.revenuecat.purchases.common.FileHelper
+import com.revenuecat.purchases.common.verboseLog
+import org.json.JSONObject
+import java.util.stream.Stream
+
+@RequiresApi(Build.VERSION_CODES.N)
+internal open class EventsFileHelper<T : Event> (
+    private val fileHelper: FileHelper,
+    private val filePath: String,
+    private val stringToEventConverter: ((String) -> T)? = null,
```

Comment:
Had to make this nullable since for diagnostics we don't parse the file back to `DiagnosticEntry` objects, as we do for `PaywallStoredEvent`. So had to add another read method that reads the file as JSONObject. 

The alternative was to make diagnostics use kotlin serialization, but as I started on that, I noticed I would have to rewrite a lot of things and didn't want to spend a lot of time there.

Reply:
Maybe
```suggestion
    private val eventDeserializer: ((String) -> T)? = null,
```

And I would leave a comment for how this behaves if it's `null`.